src/v0/sources/shopify/util.js

```diff
@@ -169,7 +169,16 @@ const setAnonymousIdorUserIdFromDb = async (message, metricMetadata) => {
   }
   let anonymousIDfromDB;
   const executeStartTime = Date.now();
-  const redisVal = await RedisDB.getVal(`${cartToken}`);
+  let redisVal;
+  try {
+    redisVal = await RedisDB.getVal(`${cartToken}`);
+  } catch (e) {
+    stats.increment('shopify_events_lost_due_redis', {
+      ...metricMetadata,
+      timestamp: Date.now(),
+    });
+    redisVal = null;
+  }
   stats.timing('redis_get_latency', executeStartTime, {
```

Comment:
We should apply same logic mentioned below for `redis_*_latency` & change the stat name to `redis_latency` & add the operation as `get` or `set` or `delete` etc., which makes things very easy for us!
Please put some thought into this and let me know what you think
https://github.com/rudderlabs/rudder-transformer/pull/2107#issuecomment-1530963143

Reply:
Makes sense updating the changes

Reply:
Done.